apps/confidential/confidential.go

```diff
@@ -400,16 +400,60 @@ func (cca Client) AuthCodeURL(ctx context.Context, clientID, redirectURI string,
 	return cca.base.AuthCodeURL(ctx, clientID, redirectURI, scopes, cca.base.AuthParams)
 }
 
+// WithTenantID specifies a tenant for a single authentication. It may be different than the tenant set in [New] by [WithAuthority].
+func WithTenantID(tenantID string) interface {
+	AcquireByAuthCodeOption
+	AcquireByCredentialOption
+	AcquireOnBehalfOfOption
+	AcquireSilentOption
+	shared.CallOption
+} {
+	return struct {
+		AcquireByAuthCodeOption
+		AcquireByCredentialOption
+		AcquireOnBehalfOfOption
+		AcquireSilentOption
+		shared.CallOption
+	}{
+		CallOption: shared.NewCallOption(
```

Comment:
I would add another restriction here, i.e. we should not allow the `tenantID` to be `common` or `organizations` because the intent of this option is to help multi-tenant apps get tokens for specific tenants only. It's ok to go from general to specific, i.e. authority = "l.m.o/common" and tenant = "12345-..", but there aren't any scenarios where you'd want to go from "common" to "common" or from "12345" to "common".

Reply:
Also, both in client_credentials flow and in OBO flow, calls should not be made over "common" / "organizations". 

- in client_credentials we should outright ban it, because "common" is only supposed to work with user flows
- in OBO, if you've got a multi-tenant app, the tenant should be `incomming_assertion.tid`

Reply:
"common" is the default confidential client tenant here, so I opened #348 to track addressing your first point. For OBO, you mean that the application should request a token from the same tenant the user authenticated in i.e., the value of the assertion's `tid` claim?